### PR TITLE
Add implementation details for export and import

### PR DESCRIPTION
**Pull request by aexolate**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -417,6 +417,44 @@ delete each person from the filtered list using `Model#deletePerson` until the l
...
 
 
+### Export/Import Feature
+
+The `export` feature allows the data file to be saved as a `.json` file on the disk. The `import` feature allows a
+data file to be imported from the disk, replacing the current data in the application.
+
+#### Implementation
+
+The import and export commands both uses `JsonAddressBookStorage` to facilitate writing and reading of data files.
```
Comment (Chen-Kuei) [inline docs/DeveloperGuide.md:427]: should be "both **use**" instead

Comment (aexolate) [reply docs/DeveloperGuide.md]: fixed!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -417,6 +417,44 @@ delete each person from the filtered list using `Model#deletePerson` until the l
...
+such as if it contains illegal value, or if no file is found at the file path.
+
+The export function calls `JsonAddressBookStorage#saveAddressBook` which will serialize a `ReadOnlyAddressBook` and
+write it to the file path on the disk.
+
+A GUI option for importing and exporting was also implemented to allow users who are unfamiliar with file paths to use
+the feature as well.
+
+Another implementation detail is creating an abstract class `FileCommand` which inherits `Command`. Both `ImportCommand`
+and `ExportCommand` inherits `FileCommand`. The reason for creating the `FileCommand` class is to reduce code
```
Comment (Chen-Kuei) [inline docs/DeveloperGuide.md:440]: both **inherit**

Comment (aexolate) [reply docs/DeveloperGuide.md]: fixed!

**Review by Chen-Kuei — APPROVED** [review]

**Pull request merged**